packages/desktop-client/src/components/reports/reports/CustomReportListCards.tsx

```diff
@@ -61,20 +63,57 @@ export function CustomReportListCards({
 }) {
   const result: { [key: string]: boolean }[] = index(reports);
   const [reportMenu, setReportMenu] = useState(result);
+  const [nameMenuOpen, setNameMenuOpen] = useState(result);
+  const [err, setErr] = useState('');
+  const [name, setName] = useState('');
+  const inputRef = createRef<HTMLInputElement>();
 
   const [isCardHovered, setIsCardHovered] = useState('');
 
+  const onAddUpdate = async ({
+    reportData,
+  }: {
+    reportData?: CustomReportEntity;
+  }) => {
+    if (!reportData) {
+      return null;
+    }
+
+    const updatedReport = {
+      ...reportData,
+      name,
+    };
+
+    const response = await sendCatch('report/update', updatedReport);
+
+    if (response.error) {
+      setErr(response.error.message);
+      onNameMenuOpen(reportData.id === undefined ? '' : reportData.id, true);
+      return;
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏we can remove these lines to reduce the amount of duplication
```suggestion
```

Reply:
This is not dupliacted. The state "true" keeps the menu open so the user can read the error. With this removed the error gets written to a menu that then immediately gets closed without time for the user to read it.